pkg/tbtc/wallet.go

```diff
@@ -260,6 +432,92 @@ func DetermineWalletMainUtxo(
 	return nil, fmt.Errorf("main UTXO not found")
 }
 
+// EnsureWalletSyncedBetweenChains makes sure all actions taken by the wallet
+// on the Bitcoin chain are reflected in the host chain Bridge. This translates
+// to two conditions that must be met:
+// - The wallet main UTXO registered in the host chain Bridge comes from the
+//   latest BTC transaction OR wallet main UTXO is unset and wallet's BTC
+//   transaction history is empty. This condition ensures that all expected SPV
+//   proofs of confirmed BTC transactions were submitted to the host chain Bridge
+//   thus the wallet state held known to the Bridge matches the actual state
+//   on the BTC chain.
+// - There are no pending BTC transactions in the mempool. This condition
+//   ensures the wallet doesn't currently perform any action on the BTC chain.
+//   Such a transactions indicate a possible state change in the future
+//   but their outcome cannot be determined at this stage so, the wallet
+//   should not perform new actions at the moment.
+func EnsureWalletSyncedBetweenChains(
+	walletPublicKeyHash [20]byte,
+	walletMainUtxo *bitcoin.UnspentTransactionOutput,
+	btcChain bitcoin.Chain,
+) error {
+	// Take the recent transactions history for the wallet.
+	history, err := btcChain.GetTransactionsForPublicKeyHash(walletPublicKeyHash, 5)
```

Comment:
I know it's just a code we moved but... why do we take 5 transactions and not just one?

Reply:
Just in case. This function tries to be universal and must find the actual main UTXO even if it doesn't come from the recent transaction.

Reply:
Just in case :D